modules/worker_thread/worker_thread.c

```diff
@@ -102,7 +102,9 @@ void* worker_thread_task_get_user_context(struct worker_thread_timer_task_s* tas
 
 #ifdef MODULE_PUBSUB_ENABLED
 void worker_thread_add_listener_task(struct worker_thread_s* worker_thread, struct worker_thread_listener_task_s* task, struct pubsub_topic_s* topic, pubsub_message_handler_func_ptr handler_cb, void* handler_cb_ctx) {
+    chSysLock();
     chDbgCheck(!worker_thread_listener_task_is_registered_I(worker_thread, task));
+    chSysUnlock();
```

Comment:
This isn't quite right because the chSysLock and chSysUnlock will be executed even if the debug checks are disabled. I've pushed a fix to my PR that should be correct. Please pull that one.